--- a/configure.ac	2016-05-22 19:39:13.000000000 +0000
+++ b/configure.ac	2016-12-02 09:05:22.692000000 +0000
@@ -74,10 +74,6 @@
     AC_DEFINE(MIO_DEBUG,1,[Define to 1 if you want to enable managed IO debug output.])
 fi
 
-# Colorized build output
-AX_CHECK_COMPILE_FLAG([-fdiagnostics-color], [CFLAGS="${CFLAGS} -fdiagnostics-color"],
-    [AX_CHECK_COMPILE_FLAG([-fcolor-diagnostics], [CFLAGS="${CFLAGS} -fcolor-diagnostics"])])
-
 # Two-step header checking.  First check for headers which don't
 # require any other headers.
 AC_HEADER_DIRENT
