diff --git a/tests/test_build_latex.py b/tests/test_build_latex.py
index 55f48cd..78ccecd 100644
--- a/tests/test_build_latex.py
+++ b/tests/test_build_latex.py
@@ -1009,7 +1009,7 @@ def test_latex_raw_directive(app, status, warning):
     assert 'HTML: abc  ghi' in result
     assert 'LaTeX: abc def ghi' in result
 
-
+@pytest.mark.xfail(reason="this test requires internet connection")
 @pytest.mark.sphinx('latex', testroot='images')
 def test_latex_remote_images(app, status, warning):
     app.builder.build_all()
