commit 1ba900a1c6a0a6d4dc3c2fd7a5de3027ecb3a153
Author: Valentina Mukhamedzhanova <vmukhame@redhat.com>
Date:   Wed Jul 1 16:17:20 2015 +0200

    Fix lvm API calls. Patch by Marek Marusic. BZ#1233152

diff --git a/yum/fssnapshots.py b/yum/fssnapshots.py
index 9af252d..a07271d 100755
--- a/yum/fssnapshots.py
+++ b/yum/fssnapshots.py
@@ -25,21 +25,21 @@ except:
 
 
 def _is_origin(lv):
-    snap = lv.getProperty("lv_attr")
+    snap = lv.getAttr()
     # snap=(<value>, <is settable>)
     if not snap[0]: # Broken??
         return None
     return snap[0][0] in ('o', 'O')
 
 def _is_snap(lv):
-    snap = lv.getProperty("lv_attr")
+    snap = lv.getAttr()
     # snap=(<value>, <is settable>)
     if not snap[0]: # Broken??
         return None
     return snap[0][0] in ('s', 'S')
 
 def _is_virt(lv):
-    snap = lv.getProperty("lv_attr")
+    snap = lv.getAttr()
     # snap=(<value>, <is settable>)
     if not snap[0]: # Broken??
         return None
