commit 6c9e839ab7ebee6357bbac8ab8ec143fc8ead461
Author: Valentina Mukhamedzhanova <vmukhame@redhat.com>
Date:   Fri May 13 13:50:14 2016 +0200

    Mention subscription-manager for enabling repos.

diff --git a/yumcommands.py b/yumcommands.py
index 1d0f5ac..2c8cecd 100644
--- a/yumcommands.py
+++ b/yumcommands.py
@@ -262,7 +262,10 @@ def checkEnabledRepo(base, possible_local_files=[]):
 
     msg = _('There are no enabled repos.\n'
             ' Run "yum repolist all" to see the repos you have.\n'
-            ' You can enable repos with yum-config-manager --enable <repo>')
+            ' To enable Red Hat Subscription Management repositories:\n'
+            '     subscription-manager repos --enable <repo>\n'
+            ' To enable custom repositories:\n'
+            '     yum-config-manager --enable <repo>')
     base.logger.critical(msg)
     raise cli.CliError
 
