commit 4474b17efc7acaa57217389cccdc36d706fdfae9
Author: Valentina Mukhamedzhanova <vmukhame@redhat.com>
Date:   Fri May 6 11:08:14 2016 +0200

    Fix a typo in exclude_updates().

diff --git a/yum/updateinfo.py b/yum/updateinfo.py
index dad6996..2b39330 100644
--- a/yum/updateinfo.py
+++ b/yum/updateinfo.py
@@ -407,7 +407,7 @@ def exclude_updates(base, filters=None):
     upds = base.doPackageLists(pkgnarrow='obsoletes')
     tot += len(upds.obsoletes)
 
-    pkgs = conduit.getPackages()
+    pkgs = base.pkgSack.returnPackages()
     name2tup = _get_name2oldpkgtup(base)
     
     cnt = 0
