diff -up yum-3.4.3/yum/updateinfo.py.old yum-3.4.3/yum/updateinfo.py
--- yum-3.4.3/yum/updateinfo.py.old	2016-03-22 12:12:51.413858074 +0100
+++ yum-3.4.3/yum/updateinfo.py	2016-03-22 12:14:56.392798309 +0100
@@ -411,13 +411,17 @@ def exclude_updates(base, filters=None):
     name2tup = _get_name2oldpkgtup(base)
     
     cnt = 0
+    pkgs_to_del = []
     for pkg in pkgs:
         name = pkg.name
         if (name not in name2tup or
             not _ysp_should_keep_pkg(opts, name2tup[name], md_info, used_map)):
-            ysp_del_pkg(pkg)
+            pkgs_to_del.append(pkg.name)
             continue
         cnt += 1
+    if pkgs_to_del:
+        for p in base.doPackageLists(pkgnarrow='available', patterns=pkgs_to_del, showdups=True).available:
+            ysp_del_pkg(p)
 
     _ysp_chk_used_map(used_map, lambda x: base.verbose_logger.warn("%s", x))
 
