commit 5a836edd592fc0adf2e1bb3883387e48cf77f548
Author: Michal Domonkos <mdomonko@redhat.com>
Date:   Mon Jan 2 12:40:46 2017 +0100

    Include repo-id in repomd.xml timestamp error. BZ 1389816

diff --git a/yum/yumRepo.py b/yum/yumRepo.py
index c6bed82..419545d 100644
--- a/yum/yumRepo.py
+++ b/yum/yumRepo.py
@@ -1378,10 +1378,11 @@ Insufficient space in download directory %s
 
         if (self.timestamp_check and
             old_repo_XML.timestamp > self.repoXML.timestamp):
-            logger.warning("Not using downloaded repomd.xml because it is "
+            logger.warning("Not using downloaded %s/repomd.xml because it is "
                            "older than what we have:\n"
                            "  Current   : %s\n  Downloaded: %s" %
-                           (time.ctime(old_repo_XML.timestamp),
+                           (self.id,
+                            time.ctime(old_repo_XML.timestamp),
                             time.ctime(self.repoXML.timestamp)))
             return False
         return True
