commit 4596a105fefc769d1d6df547f3bd172a53bf7a53
Author: Ville Skyttä <ville.skytta@iki.fi>
Date:   Fri Jan 24 16:34:35 2014 +0200

    s/ouput/output/ typo fixes

diff --git a/docs/sphinxdocs/rstgenerator.py b/docs/sphinxdocs/rstgenerator.py
index ad24788..4a0bca0 100755
--- a/docs/sphinxdocs/rstgenerator.py
+++ b/docs/sphinxdocs/rstgenerator.py
@@ -12,7 +12,7 @@ def generateFile(input_directory, file_name, output_directory,
        source code file
     :param file_name: the name of the python source code file to generate
        a sphinx rst file describing
-    :param ouput_directory: a string specifying the directory where
+    :param output_directory: a string specifying the directory where
        the generated rst file should be placed.  If *output_directory* does
        not already exist, it will be created
     :param package_heirarchy: a list of strings, where each name is
diff --git a/etc/yum-cron-hourly.conf b/etc/yum-cron-hourly.conf
index 7871a46..2a588cd 100644
--- a/etc/yum-cron-hourly.conf
+++ b/etc/yum-cron-hourly.conf
@@ -42,7 +42,7 @@ emit_via = stdio
 
 # The width, in characters, that messages that are emitted should be
 # formatted to.
-ouput_width = 80
+output_width = 80
 
 
 [email]
diff --git a/etc/yum-cron.conf b/etc/yum-cron.conf
index b0f7839..7314fae 100644
--- a/etc/yum-cron.conf
+++ b/etc/yum-cron.conf
@@ -42,7 +42,7 @@ emit_via = stdio
 
 # The width, in characters, that messages that are emitted should be
 # formatted to.
-ouput_width = 80
+output_width = 80
 
 
 [email]
diff --git a/shell.py b/shell.py
index 2232b03..00b6896 100644
--- a/shell.py
+++ b/shell.py
@@ -171,7 +171,7 @@ class YumShell(cmd.Cmd):
     def do_help(self, arg):
         """Output help information.
 
-        :param arg: the command to ouput help information about. If
+        :param arg: the command to output help information about. If
            *arg* is an empty string, general help will be output.
         """
         msg = """
