diff -uNrp yum-3.4.3.orig/yum/constants.py yum-3.4.3/yum/constants.py
--- yum-3.4.3.orig/yum/constants.py	2018-01-30 11:17:56.961899488 +0000
+++ yum-3.4.3/yum/constants.py	2018-01-30 11:41:57.421866122 +0000
@@ -123,22 +123,22 @@ REPO_PROBLEM_PACKAGE=5
 
 
 ERRORS_TO_KBASE_ARTICLES = {
-    404: """To address this issue please refer to the below knowledge base article 
+    404: """To address this issue please refer to the below wiki article 
 
-https://access.redhat.com/articles/1320623
+https://wiki.centos.org/yum-errors
 
-If above article doesn't help to resolve this issue please open a ticket with Red Hat Support.
+If above article doesn't help to resolve this issue please use https://bugs.centos.org/.
 """,
-    403: """To address this issue please refer to the below knowledge base article
+    403: """To address this issue please refer to the below wiki article
 
-https://access.redhat.com/solutions/69319
+https://wiki.centos.org/yum-errors
 
-If above article doesn't help to resolve this issue please open a ticket with Red Hat Support.
+If above article doesn't help to resolve this issue please use https://bugs.centos.org/.
 """,
-    60: """It was impossible to connect to the Red Hat servers.
+    60: """It was impossible to connect to the CentOS servers.
 This could mean a connectivity issue in your environment, such as the requirement to configure a proxy,
 or a transparent proxy that tampers with TLS security, or an incorrect system clock.
-Please collect information about the specific failure that occurs in your environment,
-using the instructions in: https://access.redhat.com/solutions/1527033 and open a ticket with Red Hat Support.
+You can try to solve this issue by using the instructions on https://wiki.centos.org/yum-errors
+If above article doesn't help to resolve this issue please use https://bugs.centos.org/.
 """
-}
\ No newline at end of file
+}
