commit d92490c2b404dab906a770a57cbdba7601eb06e6
Author: Tomas Mlcoch <tmlcoch@redhat.com>
Date:   Tue Apr 8 11:01:49 2014 +0200

    Create indexes on tables with weak deps

diff --git a/db.c b/db.c
index ae9552c..0191b06 100644
--- a/db.c
+++ b/db.c
@@ -469,7 +469,8 @@ yum_db_index_primary_tables (sqlite3 *db, GError **err)
         return;
     }
 
-    const char *deps[] = { "requires", "provides", "conflicts", "obsoletes", NULL };
+    const char *deps[] = { "requires", "provides", "conflicts", "obsoletes",
+			   "suggests", "enhances", "recommends", "supplements", NULL };
     int i;
 
     const char *pkgindexsql = "CREATE INDEX IF NOT EXISTS pkg%s on %s (pkgKey)";
